scrapy/exporters.py

```diff
@@ -86,10 +88,10 @@ def __init__(self, file, **kwargs):
 
     def export_item(self, item):
         itemdict = dict(self._get_serialized_fields(item))
-        self.file.write(self.encoder.encode(itemdict) + '\n')
+        self.file.write(to_bytes(self.encoder.encode(itemdict) + '\n'))
 
 
-class JsonItemExporter(JsonLinesItemExporter):
+class JsonItemExporter(BaseItemExporter):
```

Comment:
I think it makes more sense to have them both as subclasses of `BaseItemExporter` than inheriting `JsonItemExporter` from `JsonLinesItemExporter` and overriding *all* methods. It is also not is-a relationship. There is a bit of code duplication in `export_item` and `__init__`, but it's not that bad.